src/sagemaker/serve/utils/hardware_detector.py

```diff
@@ -0,0 +1,110 @@
+    "ml.p3dn.24xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 262144},
+    "ml.p2.xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 12288},
+    "ml.p2.8xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 98304},
+    "ml.p2.16xlarge": {"Count": 16, "TotalGpuMemoryInMiB": 196608},
+    "ml.g4dn.xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 16384},
+    "ml.g4dn.2xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 16384},
+    "ml.g4dn.4xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 16384},
+    "ml.g4dn.8xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 16384},
+    "ml.g4dn.16xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 16384},
+    "ml.g4dn.12xlarge": {"Count": 4, "TotalGpuMemoryInMiB": 65536},
+    "ml.g5n.xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 22888},
+    "ml.g5.2xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 24576},
+    "ml.g5.4xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 24576},
+    "ml.g5.8xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 24576},
+    "ml.g5.16xlarge": {"Count": 1, "TotalGpuMemoryInMiB": 24576},
+    "ml.g5.12xlarge": {"Count": 4, "TotalGpuMemoryInMiB": 98304},
+    "ml.g5.24xlarge": {"Count": 4, "TotalGpuMemoryInMiB": 98304},
+    "ml.g5.48xlarge": {"Count": 8, "TotalGpuMemoryInMiB": 196608},
+}
+
+
+def _get_gpu_info(instance_type: str, session: Session) -> tuple[int, int]:
+    """Get GPU info for the provided instance
+
+    @return: Tuple containing: [0]number of GPUs available and [1]aggregate memory size in MiB
+    """
+    ec2_client = session.boto_session.client("ec2")
+    ec2_instance = _format_instance_type(instance_type)
+
+    try:
+        instance_info = ec2_client.describe_instance_types(InstanceTypes=[ec2_instance]).get(
+            "InstanceTypes"
+        )[0]
+    except ClientError as e:
+        logger.error("Failed to describe instance types %s", e)
+        raise ValueError("Provided instance_type is not GPU enabled.")
+
+    if instance_info is not None:
+        gpus_info = instance_info.get("GpuInfo")
+        if gpus_info is not None:
+            instance_gpu_info = (
+                gpus_info.get("Gpus")[0].get("Count"),
+                gpus_info.get("TotalGpuMemoryInMiB"),
+            )
+
+            logger.info("GPU Info [%s]: %s", ec2_instance, instance_gpu_info)
+            return instance_gpu_info
+
+    logger.error("Provided instance_type is not GPU enabled: [%s]", ec2_instance)
+    raise ValueError("Provided instance_type is not GPU enabled.")
+
+
+def _get_gpu_info_fallback(instance_type: str) -> tuple[int, int]:
+    """Get GPU info for the provided instance fallback
+
+    @return: Tuple containing: [0]number of GPUs available and [1]aggregate memory size in MiB
+    """
+    fallback_instance_gpu_info = FALLBACK_GPU_RESOURCE_MAPPING.get(instance_type)
+    ec2_instance = _format_instance_type(instance_type)
+    if fallback_instance_gpu_info is None:
+        logger.error("Provided instance_type is not GPU enabled: [%s]", ec2_instance)
```

Comment:
Same just pass this message through the ValueError, we don't log and throw. We log or Throw...